subprojects/platform-native/src/test/groovy/org/gradle/nativeplatform/toolchain/internal/swift/SwiftLinkerTest.groovy

```diff
@@ -0,0 +1,219 @@
+import org.gradle.nativeplatform.platform.internal.DefaultOperatingSystem
+import org.gradle.nativeplatform.toolchain.internal.CommandLineToolContext
+import org.gradle.nativeplatform.toolchain.internal.CommandLineToolInvocation
+import org.gradle.nativeplatform.toolchain.internal.CommandLineToolInvocationWorker
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.gradle.test.fixtures.work.TestWorkerLeaseService
+import org.gradle.util.UsesNativeServices
+import org.junit.Rule
+import spock.lang.Specification
+
+@UsesNativeServices
+class SwiftLinkerTest extends Specification {
+    public static final String LOG_LOCATION = "<log location>"
+    @Rule final TestNameTestDirectoryProvider tmpDirProvider = new TestNameTestDirectoryProvider()
+
+    def operationLogger =  Mock(BuildOperationLogger)
+    def executable = new File("executable")
+    def invocationContext = Mock(CommandLineToolContext)
+    def invocation = Mock(CommandLineToolInvocation)
+    CommandLineToolInvocationWorker commandLineTool = Mock(CommandLineToolInvocationWorker)
+    BuildOperationExecutor buildOperationExecutor = Mock(BuildOperationExecutor)
+    BuildOperationQueue queue = Mock(BuildOperationQueue)
+    WorkerLeaseService workerLeaseService = new TestWorkerLeaseService()
+
+    SwiftLinker linker = new SwiftLinker(buildOperationExecutor, commandLineTool, invocationContext, workerLeaseService)
+
+    def "ignores install name for all major operating system"() {
+        given:
+        def testDir = tmpDirProvider.testDirectory
+        def outputFile = testDir.file("output/lib")
+
+        final expectedArgs = [
+            "-emit-library",
+            "-o", outputFile.absolutePath,
+            testDir.file("one.o").absolutePath].flatten()
+
+        when:
+        NativePlatform platform = Mock(NativePlatform)
+        platform.getOperatingSystem() >> new DefaultOperatingSystem(operatingSystem.name, operatingSystem)
+
+        LinkerSpec spec = Mock(SharedLibraryLinkerSpec)
+        spec.getSystemArgs() >> []
+        spec.getArgs() >> []
+        spec.getOutputFile() >> outputFile
+        spec.getLibraries() >> []
+        spec.getLibraryPath() >> []
+        spec.getInstallName() >> "installName"
+        spec.getTargetPlatform() >> platform
+        spec.getObjectFiles() >> [testDir.file("one.o")]
+        spec.getOperationLogger() >> operationLogger
+
+        and:
+        linker.execute(spec)
+
+        then:
+        1 * operationLogger.getLogLocation() >> LOG_LOCATION
+        1 * buildOperationExecutor.runAll(commandLineTool, _) >> { worker, action -> action.execute(queue) }
+        1 * invocationContext.getArgAction() >> Actions.doNothing()
+        1 * invocationContext.createInvocation("linking lib", outputFile.parentFile, expectedArgs, operationLogger) >> invocation
+        1 * queue.add(invocation)
+        1 * queue.setLogLocation(LOG_LOCATION)
+        0 * _
```

Comment:
**Open questions:**
- How to remove this code duplication between tests?

Reply:
Have fewer collaborators. There's too much stuff going on in the task that could live behind a service of some kind (for example the `Compiler` implementations).